cluster-autoscaler/ca.patch

```diff
@@ -1,3 +1,103 @@
+ 				Pods: eg.Pods,
+diff --git a/cluster-autoscaler/estimator/binpacking_estimator.go b/cluster-autoscaler/estimator/binpacking_estimator.go
+index 229a2e264..f145a218f 100644
+--- a/cluster-autoscaler/estimator/binpacking_estimator.go
++++ b/cluster-autoscaler/estimator/binpacking_estimator.go
+@@ -108,13 +108,18 @@ func (e *BinpackingNodeEstimator) Estimate(
+ 		var err error
+ 		var remainingPods []*apiv1.Pod
+ 
++		klog.V(4).Infof("trying to schedule %d pods on existing nodes", len(podsEquivalenceGroup.Pods))
+ 		remainingPods, err = e.tryToScheduleOnExistingNodes(estimationState, podsEquivalenceGroup.Pods)
++
++		klog.V(4).Infof("After attempt, there are %d pods that couldn't be scheduled", len(remainingPods))
+ 		if err != nil {
+ 			klog.Errorf(err.Error())
+ 			return 0, nil
+ 		}
+ 
++		klog.V(4).Infof("trying to schedule %d pods on new nodes", len(remainingPods))
+ 		err = e.tryToScheduleOnNewNodes(estimationState, nodeTemplate, remainingPods)
++		klog.V(4).Infof("After attempt, there are %d pods that couldn't be scheduled", len(remainingPods))
+ 		if err != nil {
+ 			klog.Errorf(err.Error())
+ 			return 0, nil
+@@ -137,6 +142,7 @@ func (e *BinpackingNodeEstimator) tryToScheduleOnExistingNodes(
+ 
+ 		// Check schedulability on all nodes created during simulation
+ 		nodeName, err := e.predicateChecker.FitsAnyNodeMatching(e.clusterSnapshot, pod, func(nodeInfo *schedulerframework.NodeInfo) bool {
++			klog.V(4).Infof("checking if pod %s.%s with nodeSelector %v fits on node %s", pod.Namespace, pod.Name, pod.Spec.NodeSelector, nodeInfo.Node().Name)
+ 			return estimationState.newNodeNames[nodeInfo.Node().Name]
+ 		})
+ 		if err != nil {
+@@ -144,6 +150,7 @@ func (e *BinpackingNodeEstimator) tryToScheduleOnExistingNodes(
+ 		}
+ 
+ 		if err := e.tryToAddNode(estimationState, pod, nodeName); err != nil {
++			klog.V(4).Infof("Error while adding pod %s.%s to node %s in ClusterSnapshot; %v", pod.Namespace, pod.Name, nodeName, err)
+ 			return nil, err
+ 		}
+ 	}
+@@ -183,6 +190,7 @@ func (e *BinpackingNodeEstimator) tryToScheduleOnNewNodes(
+ 			// each call that returns true, one node gets added. Therefore this
+ 			// must be the last check right before really adding a node.
+ 			if !e.limiter.PermissionToAddNode() {
++				klog.V(4).Infof("Tried to add node, but couldn't due to hitting the limit")
+ 				break
+ 			}
+ 
+@@ -195,10 +203,12 @@ func (e *BinpackingNodeEstimator) tryToScheduleOnNewNodes(
+ 			// Note that this may still fail (ex. if topology spreading with zonal topologyKey is used);
+ 			// in this case we can't help the pending pod. We keep the node in clusterSnapshot to avoid
+ 			// adding and removing node to snapshot for each such pod.
+-			if err := e.predicateChecker.CheckPredicates(e.clusterSnapshot, pod, estimationState.lastNodeName); err != nil {
++			err := e.predicateChecker.CheckPredicates(e.clusterSnapshot, pod, estimationState.lastNodeName)
++			if err != nil && err.PredicateName() != "InterPodAffinity" {
```

Comment:
question(blocking): Should this be `if err != nil || err.PredicateName() == "InterPodAffinity"` ? or if not, what's the purpose of this change?

\+ similar modification w.r.t. also checking against the error message

Reply:
Yeah, good point.